base-notebook/start.sh

```diff
@@ -45,9 +45,23 @@ if [ $(id -u) == 0 ] ; then
         echo "$NB_USER ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/notebook
     fi
 
-    # Exec the command as NB_USER
+    # Exec the command as NB_USER or root
     echo "Execute the command: $*"
-    exec su $NB_USER -c "env PATH=$PATH $*"
+    if [[ "$GRANT_SUDO" == "1" || "$GRANT_SUDO" == 'yes' ]]; then
+        if [ $# -eq 0 ]; then
+            echo "No arguments supplied"
+            sudo -H -u $NB_USER bash -c 'env; PATH=$PATH; bash'
+        else
+            exec su $NB_USER -c "env PATH=$PATH $*"
+        fi
+    else
+        if [ $# -eq 0 ]; then
+            echo "No arguments supplied"
+            bash -c 'env; PATH=$PATH; bash'
+        else
+            env PATH=$PATH $*
```

Comment:
The test failure is due to the user never becoming the appropriate `NB_USER` in the case where the container is run as root to configure an alternative `NB_UID` or `NB_GID` without `GRANT_SUDO` set. The original `start.sh` logic used the `exec su $NB_USER` command regardless of whether `GRANT_SUDO` was set or not to get the correct behavior here.

I think using the `exec su` form on line 62 should fix that problem, and will make that change after verifying the tests pass locally and refactoring the duplicate script lines. I'll need to toy with line 60 some to see how that behaves in the various sudo/no-sudo, root-on-start/no-root-on-start, etc. configurations. Some additional tests will probably help here.

Sidenote: yay for tests!